src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -3839,14 +4165,22 @@ private static string GetTimeZoneDirectory()
             for (int i = 0; i < t.IsGmtCount && i < t.TypeCount && index < data.Length; i++) {
                 GmtTime[i] = (data[index++] != 0);
             }
-        }
 
+            if (t.Version != TZVersion.V1)
+            {
+                // read the POSIX-style format, which should be wrapped in newlines with the last newline at the end of the file
+                if (data[index++] == '\n' && data[data.Length - 1] == '\n')
+                {
+                    futureTransitionsPosixFormat = enc.GetString(data, index, data.Length - index - 1);
+                }
+            }
+        }
 
         // Windows NT time is specified as the number of 100 nanosecond intervals since January 1, 1601.
         // UNIX time is specified as the number of seconds since January 1, 1970. There are 134,774 days
         // (or 11,644,473,600 seconds) between these dates.
         //
-        private static DateTime TZif_UnixTimeToWindowsTime(int unixTime) {
+        private static DateTime TZif_UnixTimeToWindowsTime(long unixTime) {
```

Comment:
Looks like this may now be duplication of DateTimeOffset.FromUnixTimeSeconds/Milliseconds?